File: Lua/PrimeFinder/prime.lua
print("Finding the first 1000 prime numbers")

function isPrime(n)
    if n < 2 then
        return false
    end
    
    local i = 2
    
    while i * i <= n do
        if n % i == 0 then
            return false
        end
        i = i + 1
    end

    return true
end

local prime_count = 0
local i = 1
while prime_count < 1000 do
    if isPrime(i) then
        prime_count = prime_count + 1
        print(i)
    end
    i = i + 1
end

File: Scala/PrimeFinder/Prime.scala
object Prime {

    def is_prime(i: Int): Boolean = {
        if (i < 2) {
            return false
        }
        var n: Int = 2
        while((n*n) <= i){
            if(i%n == 0){
                return false
            }
            n += 1
        }

        return true
    }

    def main(args: Array[String]) = {
        println("Finding the First 1000 Prime Numbers.")
        var Prime_Count: Int = 0
        var i: Int = 1

        while(Prime_Count < 1000){
            if(is_prime(i)){
                Prime_Count += 1
                println(i)
            }
            i += 1
        }
    }
}